classes/class-wc-rest-connect-post-print-notification-settings-controller.php

```diff
@@ -0,0 +1,34 @@
+<?php
```

Comment:
I see we use `classes/class-wc-rest-connect-shipping-label-preview-controller.php` (`connect/label/preview`) when we save the `paper_size` preference. I am a little surprised that I can't find an existing end point to update settings. So this new end point makes sense.

Were you also not able to find any existing end point to update settings?

Reply:
Yes!

I was considering adding the notification-settings-saving logic to an existing endpoint but:

1. Saving when shipping rates are requested or purchased is too early because the checkbox might be toggled afterward.
1. Saving when the label PDF is **requested** (this happens automatically after a request to purchase a label is made) was another option I considered but:
    1. the notifications are not yet sent at this point so again, this is too early;
    1. this is a `GET` request. The paper size setting is saved upon this request, but this makes more sense than saving notification settings here - the paper size has to be provided anyway so a label can be printed in the selected size.
1. Saving when the label PDF is **displayed** (either after "Print" is clicked or the PDF is opened automatically) *sometimes* triggers another request. This will be either a request to mark an order as completed, to send the email with details, but can also be no request at all. In such cases, a separate call would have to be made anyway.

Based on the above, I decided that placing the logic in a separate REST endpoint accessed after the "Print" button is clicked would be the cleanest solution.

Reply:
This makes a lot of sense now. Thank you!